### PR TITLE
AEA-1837 Fix Test Pack examples in EPSAT

### PR DESCRIPTION
**Pull request by booshibooshi**

### CODE REVIEW

File: tool/site/client/parsers/read/medication-request-parser.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import {MedicationRequest} from "../../models"
 import * as fhirExtension from "../../models/extension"
 
 export function getMedicationRequestNumberOfRepeatsAllowed(medicationRequest: MedicationRequest): string | number {
-  const repeatInformation = medicationRequest.extension.find(e =>
+  const repeatInformation = medicationRequest.dispenseRequest?.extension.find(e =>
```
Comment (mfjarvis) [inline tool/site/client/parsers/read/medication-request-parser.ts:5]: This extension is still in `MedicationRequest.extension`. It's just numberOfRepeatsAllowed which has moved to `MedicationRequest.dispenseRequest.numberOfRepeatsAllowed`. See `getRepeatsIssuedAndAllowed()` in `tool/site/client-react/src/fhir/helpers.ts`

File: tool/site/client/test-pack/prescriptions.ts [hunk 2]
```diff
@@ -74,7 +74,7 @@ function createRepeatPrescribingPrescriptions(
   prescriptions: any[]
 ) {
   const repeatsAllowed = getNumberOfRepeatsAllowed(prescriptionRow)
-  for (let repeatsIssued = 0; repeatsIssued < repeatsAllowed; repeatsIssued++) {
+  for (let repeatsIssued = 1; repeatsIssued <= repeatsAllowed; repeatsIssued++) {
```
Comment (mfjarvis) [inline tool/site/client/test-pack/prescriptions.ts:77]: Looks like we're sending in a message for each issue of the prescription, but that's not how eRD works. You just send in the first issue and Spine creates all of the subsequent issues automatically

Comment (mfjarvis) [reply tool/site/client/test-pack/prescriptions.ts:77]: And for repeat prescribing, even if you're sending in issue 3 of 6 for example, you still tell Spine it's issue 1 of 1.

Comment (booshibooshi) [reply tool/site/client/test-pack/prescriptions.ts:77]: as discussed, constraints from backend means this information is ignored, but suggest populating it as we are expecting in FHIR to be consistent with our docs and not expose inner workings of backend

File: tool/site/client/parsers/read/medication-request-parser.ts (hunk 1 above)
Comment (booshibooshi) [reply tool/site/client/parsers/read/medication-request-parser.ts]: Moved

File: tool/site/client-react/src/fhir/helpers.ts [hunk 3]
```diff
@@ -65,11 +64,6 @@ export function getRepeatsIssuedAndAllowed(medicationRequest: MedicationRequest)
   const numberOfRepeatPrescriptionsIssued = ukCoreRepeatsIssuedExtension
     ? ukCoreRepeatsIssuedExtension.valueUnsignedInt
     : 1
-
-  const ukCoreRepeatsAllowedExtension = getUkCoreNumberOfRepeatsAllowedExtension(medicationRequest.extension)
-  const numberOfRepeatPrescriptionsAllowed = ukCoreRepeatsAllowedExtension
-    ? ukCoreRepeatsAllowedExtension.valueUnsignedInt
-    : (medicationRequest.dispenseRequest.numberOfRepeatsAllowed || 0) + 1
-
+  const numberOfRepeatPrescriptionsAllowed = medicationRequest.dispenseRequest?.numberOfRepeatsAllowed || 1
```
Comment (mfjarvis) [inline tool/site/client-react/src/fhir/helpers.ts:67]: Think this should be `(medicationRequest.dispenseRequest.numberOfRepeatsAllowed || 0) + 1`

File: tool/site/client-react/src/fhir/customExtensions.ts [hunk 4]
```diff
@@ -99,7 +99,7 @@ export const getNumberOfRepeatsAllowedExtension = (extensions: Array<Extension>)
   ])
 
 const URL_UK_CORE_REPEAT_INFORMATION = "https://fhir.hl7.org.uk/StructureDefinition/Extension-UKCore-MedicationRepeatInformation"
-export const URL_UK_CORE_NUMBER_OF_REPEATS_ISSUED = "numberOfRepeatPrescriptionsIssued"
+export const URL_UK_CORE_NUMBER_OF_REPEATS_ISSUED = "numberOfPrescriptionsIssued"
 export const URL_UK_CORE_NUMBER_OF_REPEATS_ALLOWED = "numberOfRepeatPrescriptionsAllowed"
```
Comment (mfjarvis) [inline tool/site/client-react/src/fhir/customExtensions.ts:103]: Can this go too?

File: tool/site/client-react/tests/components/claim/__snapshots__/createDispenseClaim.spec.ts.snap [hunk 5]
```diff
@@ -51,7 +51,7 @@ Object {
                 },
                 Object {
                   "url": "numberOfRepeatsAllowed",
-                  "valueInteger": 6,
+                  "valueInteger": 5,
```
Comment (mfjarvis) [inline tool/site/client-react/tests/components/claim/__snapshots__/createDispenseClaim.spec.ts.snap:54]: See above - we need to add 1 to `numberOfRepeatsAllowed` when getting it from the dispenseRequest

File: tool/site/client/transformers/dispense-notification.ts [hunk 6]
```diff
@@ -177,11 +177,7 @@ function createDispensingRepeatInformationExtension(
   // const numberOfRepeatPrescriptionsIssued = numberOfRepeatPrescriptionsIssuedExtension.valueUnsignedInt
   const issueNumberStr = (document.getElementById("issue-number") as HTMLInputElement).value
   const numberOfRepeatPrescriptionsIssued = parseInt(issueNumberStr, 10)
-
-  const numberOfRepeatPrescriptionsAllowedExtension = repeatInformationExtension.extension.find(e =>
-    e.url === "numberOfRepeatPrescriptionsAllowed"
-  ) as fhirExtension.UnsignedIntExtension
-  const numberOfRepeatPrescriptionsAllowed = numberOfRepeatPrescriptionsAllowedExtension.valueUnsignedInt
+  const numberOfRepeatPrescriptionsAllowed = medicationRequest.dispenseRequest?.numberOfRepeatsAllowed || 1
```
Comment (mfjarvis) [inline tool/site/client/transformers/dispense-notification.ts:180]: Here too

**Review by mfjarvis — APPROVED** [review]

**Pull request merged**